### PR TITLE
[IAP] UI for plan benefit details screens

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #9967, follows #9973

## Description

We're adding the ability to purchase the Woo Express Essential Monthly plan, using In-App Purchase.

This updates the UI for the Upgrades screen to be closer to the designs. In particular, it:
- Adds the `WooPlanFeatureBenefitsView`, and makes each feature group row on the Upgrades screen open the relevant benefits view

~However, it's not a complete implementation of the screen. We still need to:~
- ~Tidy up a lot of constants and numbers~
- ~Review the fonts and other design details~
- ~Make it top aligned when the content is smaller than the screen, not vertically centred~

## Testing instructions

Using a free trial Woo Express store

1. Open the app
2. Tap Upgrade Now
3. Wait for the screen to load
4. Tap in to the feature benefit rows
5. Observe that the views matches the screen capture video below

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/2472348/1ee3c028-994a-4128-9b4e-a35cea7d9ce9

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by joshheald** [comment]
@iamgabrielma FYI this is how far ChatGPT and I got. I still have more tidying up to do as covered in the description, before it's really ready for review, but this should give us something to show the folks who need to review copy.

**Comment by joshheald** [comment]
@iamgabrielma This one's ready now, IMO. There's still more to do one the Upgrades screen before we close the ticket, but getting close now. Will finish that off tomorrow.

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanFeatureBenefitsView.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import SwiftUI
+import Yosemite
+import WooFoundation
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Upgrades/WooPlanFeatureBenefitsView.swift]: It seems we don't need this import for the moment:
```suggestion
```

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanFeatureBenefitsView.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    private enum Layout {
+        static let verticalPadding: CGFloat = 4
+        static let checkmarkWidth: CGFloat = 24
+        static let titleDescriptionVerticalSpacing: CGFloat = 2
+        static let checkmarkToTextHorizontalSpacing: CGFloat = 8
+    }
+}
+
+struct WooPlanFeatureBenefitsView_Previews: PreviewProvider {
+    static let featureGroup: WooPlanFeatureGroup = WooPlan()!.planFeatureGroups[0]
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Upgrades/WooPlanFeatureBenefitsView.swift:80]: We could create some sort of static method or convenience init when we need to use a WooPlan around but we do not care about its details, let's say:

```
public extension WooPlan {
    static func wooPlanFactory() -> WooPlan {
        WooPlan.init(id: "", name: "", shortName: "", planFrequency: .month, planDescription: "", headerImageFileName: "", planFeatureGroups: [])
    }
}
```
```suggestion
    static let featureGroup: WooPlanFeatureGroup = WooPlan.wooPlanFactory().planFeatureGroups[0]
```

**Review by iamgabrielma — APPROVED** [review]
Looks great! 🚢

**Pull request merged**